console/metrics.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Console Metrics for Telemetry
+authors:
+  - "@benjaminapetersen"
+reviewers:
+  - "@spadgett"
+  - "@bparees"
+approvers:
+  - "@spadgett"
+  - "@bparees"
+creation-date: 2019-09-20
+last-updated: 2019-09-20
+status: implementable
+---
+
+# Console Metrics for Telemetry
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The purpose of this enhancement is to enable a set of telemetry metrics 
+for the console in order to improve the support of our customer environments.  
+
+## Motivation
+
+### Goals
+
+- Tracking metrics via telemetry would allow us to gather feature usage data for the 
+console so that we can focus development efforts on the features customers find most 
+valuable.
+
+### Non-Goals
+
+- Analytics will not be handled in this proposal.  Previously, the console supported an optional 
+Google Analytics identifier. No other analytics platform is or has been supported.
+- Error tracking will not be handled in this proposal.  We have discussed the use of tools like 
+Sentry for collecting errors and stack traces to improve our ability to understand console 
+error frequency and severity within the browser.
+
+## Proposal
+
+- Track the console URL as a metadata metric like `console_url` to support UHC to provide backlinking 
+to the console.
+- Track API requests from the console to the API server as a metric like `console_api_requests_total` 
+with labels such as `referer` and `path` to clearly understand how the console consumes the API and 
```

Comment:
"referer" seems like it has the potential to explode cardinality (also path, since it includes the namespace, right?  so if nothing else, it wouldn't be consumable by telemeter i don't think).  @s-urbaniak @brancz are these viable label values?  i'm thinking not...

Reply:
also if we did want to gather api request metrics, i'd rather see the api server reporting who was calling it, with the client info, rather than making each client/caller report its own metrics.

Reply:
`referer` may be too generic and have that problem, agree.  Ideally we want to know what page made what request, knowing `deployments` or `deployment instance` would be ideal, I don't think we want to know full domain and path.  

So from this:

`https://console-openshift-console.apps.bpetersen.devcluster.openshift.com/k8s/ns/default/deployments/example`

Really we want to know `deployment instance page`.

Reply:
And def an interesting point on the `api server` reporting who called it.  I can see both being valuable.